auth_jwt/tests/test_auth_jwt.py

```diff
@@ -0,0 +1,250 @@
+# Copyright 2021 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import contextlib
+import time
+from unittest.mock import Mock
+
+import jwt
+
+import odoo.http
+from odoo.exceptions import ValidationError
+from odoo.tests.common import TransactionCase
+from odoo.tools import mute_logger
+from odoo.tools.misc import DotDict
+
+from ..exceptions import (
+    AmbiguousJwtValidator,
+    JwtValidatorNotFound,
+    UnauthorizedInvalidToken,
+    UnauthorizedMalformedAuthorizationHeader,
+    UnauthorizedMissingAuthorizationHeader,
+    UnauthorizedPartnerNotFound,
+)
+
+
+class TestAuthMethod(TransactionCase):
+    @contextlib.contextmanager
+    def _mock_request(self, authorization):
+        request = Mock(
+            context={},
+            db=self.env.cr.dbname,
+            uid=None,
+            httprequest=Mock(environ={"HTTP_AUTHORIZATION": authorization}),
+            session=DotDict(),
+        )
+
+        with contextlib.ExitStack() as s:
+            odoo.http._request_stack.push(request)
+            s.callback(odoo.http._request_stack.pop)
+            yield request
+
+    def _create_token(
+        self,
+        key="thesecret",
+        audience="me",
+        issuer="http://the.issuer",
+        exp_delta=100,
+        nbf=None,
+        email=None,
+    ):
+        payload = dict(aud=audience, iss=issuer, exp=time.time() + exp_delta)
+        if email:
+            payload["email"] = email
+        if nbf:
+            payload["nbf"] = nbf
+        return jwt.encode(payload, key=key, algorithm="HS256")
+
+    def _create_validator(self, name, audience="me", partner_id_required=False):
+        return self.env["auth.jwt.validator"].create(
+            dict(
+                name=name,
+                signature_type="secret",
+                secret_algorithm="HS256",
+                secret_key="thesecret",
+                audience=audience,
+                issuer="http://the.issuer",
+                user_id_strategy="static",
+                partner_id_strategy="email",
+                partner_id_required=partner_id_required,
+            )
+        )
+
+    @contextlib.contextmanager
+    def _commit_validator(self, name, audience="me", partner_id_required=False):
+        validator = self._create_validator(
+            name=name, audience=audience, partner_id_required=partner_id_required
+        )
+        # commit because IrHttp._auth_method_jwt will look for validator in another tx
+        self.env.cr.commit()  # pylint: disable=invalid-commit
```

Comment:
no way to get rid of this? Maybe mocking the lookup of the validator on ir.http?

Reply:
@simahawk I updated the code to mock instead of commit.